src/main/java/org/chocosolver/solver/constraints/nary/cumulative/literature/PropagatorCumulative.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2019, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.cumulative.literature;
+
+import org.chocosolver.solver.constraints.Propagator;
+import org.chocosolver.solver.constraints.PropagatorPriority;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.exception.SolverException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.Task;
+import org.chocosolver.solver.variables.events.IntEventType;
+import org.chocosolver.util.ESat;
+import org.chocosolver.util.tools.ArrayUtils;
+
+/**
+ * Cumulative propagator
+ * Encapsulates filtering algorithm for the cumulative and disjunctive constraints.
+ *
+ * @author Arthur Godet <arth.godet@gmail.com>
+ * @since 23/05/2019
+ */
+public class PropagatorCumulative extends Propagator<IntVar> {
```

Comment:
The class name should not be so generic, there is ann other `PropagatorCumulative` classe, that may be confusing.